### PR TITLE
12 add food consume api

### LINKED ISSUE #12: AddFood component should consume addFood(food) API function
The data created in our `AddFood` form needs to be sent through to our database when we submit the form.

-  In order to send our new data to the database we will need an `addFood(food)` API function that takes the new food added and sends it through to our POST `/api/v1/foods/` route.
-  We're going to need to use a `submitHandler` to take the component form state and pass it into our `addFood(food)` API function.

> Note that this task is being accomplished without the use of redux. Our state is local component state only and goes straight to the database on submit. This means this api will be in a seperate (`api.js`) file to the redux thunk actions.
(For further explanation please see a lead developer).

Blocked by #9

### PR DESCRIPTION
**Pull request by joanna-klimas**
closes #12

### CODE REVIEW

File: client/api/__mocks__/api.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export function addFood(food) {
```
Comment (emilyparkes) [inline client/api/__mocks__/api.js]: Not sure why this is in a mock folder but isn't mocking anything. Should this be in `tests/client/_mocks_` like this?

https://github.com/don-smith/reka/blob/master/tests/client/mocks/api.js

Comment (dieseldb) [reply client/api/__mocks__/api.js]: Done as you command

**Review by emilyparkes — DISMISSED** [review]

File: client/api/api.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import request from 'superagent'
+
+export function addFood(food) {
+  return request.post('/api/v1/foods')
+    .send(food)
+}
```
Comment (emilyparkes) [inline client/api/api.js]: New line at end of file here too please 😛

File: tests/client/components/AddFood.test.js [hunk 3]
```diff
@@ -22,8 +23,21 @@ test('handleInputChange changes state of the component', () => {
...
+  const expected = true
+
+  const app = wrapper.instance()
+  app.handleSubmit().then(() => {
+    const actual = app.state.redirect
+
+    expect(actual).toBe(expected)
+  })
+
+})
```
Comment (emilyparkes) [inline tests/client/components/AddFood.test.js]: new line 😛

File: tests/client/mocks/api.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+jest.mock('../../../client/api/api', () => {
+  return new Promise((resolve, reject) => {
+    return Promise.resolve({body: 'Post OK'})
+  })
+})
```
Comment (emilyparkes) [inline tests/client/mocks/api.js]: new line 😛

**Review by emilyparkes — CHANGES_REQUESTED** [review]

File: client/api/api.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+import request from 'superagent'
+
+export function addFood(food) {
+  return request.post('/api/v1/foods')
+    .send(food)
```
Comment (emilyparkes) [inline client/api/api.js:5]: Could we have a catch block on this?

**Review by emilyparkes — COMMENTED** [review]
Getting Errors/Warnings with promises that may be related to this PR so won't accept yet.

**Review by emilyparkes — APPROVED** [review]

**Pull request merged**